.travis.sh

```diff
@@ -0,0 +1,22 @@
+branch=$1
+if [ "x" = "x$branch" ]; then
+    echo 'Usage: branch dir1 dir2 ... dirN'
+    exit 1
+fi
+shift
+components="$*"
+if [ "x" = "x$components" ]; then
+    echo 'Usage: branch dir1 dir2 ... dirN'
+    exit 1
+fi
+echo '{"packages": {' > packages.json
+for c in $components; do
+    sed -i -z 's#^{\(\s*\)\("name"\)#{\1"repositories": \[{ "type": "composer", "url": "file://'$(pwd)'/" }\],\1\2#' $c/composer.json
```

Comment:
looks like you are adding the same composer repository 30 times here approx (once per subtree repo). This seems useless. It should be added only once outside the loop